packages/swing-store/src/snapStore.js

```diff
@@ -205,22 +241,25 @@ export function makeSnapStore(
   }
 
   const sqlLoadSnapshot = db.prepare(`
-    SELECT compressedSnapshot
+    SELECT hash, compressedSnapshot
     FROM snapshots
-    WHERE hash = ?
+    WHERE vatID = ?
+    ORDER BY endPos DESC
+    LIMIT 1
   `);
-  sqlLoadSnapshot.pluck(true);
 
   /**
-   * @param {string} hash
+   * Loads the most recent snapshot for a given vat.
+   *
+   * @param {string} vatID
    * @param {(filePath: string) => Promise<T>} loadRaw
    * @template T
    */
-  async function load(hash, loadRaw) {
+  async function load(vatID, loadRaw) {
     const cleanup = [];
     return aggregateTryFinally(
       async () => {
-        const compressedSnapshot = sqlLoadSnapshot.get(hash);
+        const { hash, compressedSnapshot } = sqlLoadSnapshot.get(vatID);
```

Comment:
It might be tidy to do `info = sqlLoadSnapshot.get(vatID); assert(info, "useful error message"); const { hash, compressedSnapshot } = info`, just to avoid the weirdness of `hash = undefined; compressedSnapshot = undefined` in case this somehow gets called when there aren't any snapshots at all. The old version was slightly less likely to be called that way because the caller had to pass in the hash.

Also, given the change to the API signature of `load()`, there's a slightly higher chance that something will call this with `hash` instead of `vatID`, and that will certainly cause the query to return an empty result.